server/src/main/java/com/linecorp/centraldogma/server/internal/storage/repository/git/GitRepository.java

```diff
@@ -829,14 +838,31 @@ private static TreeFilter pathPatternFilterOrTreeFilter(@Nullable String pathPat
                                 final JsonNode newJsonNode =
                                         Jackson.readTree(
                                                 reader.open(diffEntry.getNewId().toObjectId()).getBytes());
-                                final JsonPatch patch =
-                                        JsonPatch.generate(oldJsonNode, newJsonNode, ReplaceMode.SAFE);
+                                patch = JsonPatch.generate(oldJsonNode, newJsonNode, ReplaceMode.SAFE);
 
                                 if (!patch.isEmpty()) {
                                     putChange(changeMap, newPath,
                                               Change.ofJsonPatch(newPath, Jackson.valueToTree(patch)));
                                 }
                                 break;
+                            case YAML:
+                                if (!oldPath.equals(newPath)) {
+                                    putChange(changeMap, oldPath, Change.ofRename(oldPath, newPath));
+                                }
+
+                                final JsonNode oldYamlNode =
+                                        JacksonYaml.readTree(
+                                                reader.open(diffEntry.getOldId().toObjectId()).getBytes());
+                                final JsonNode newYamlNode =
+                                        JacksonYaml.readTree(
+                                                reader.open(diffEntry.getNewId().toObjectId()).getBytes());
```

Comment:
On second thought, it would be better to make `Jackson` interface and use like:
```java
Jackson.ofJson().readTree(...)
Jackson.ofYaml().readTree(...)
// dynamically get the implementation 
Jackson.of(YAML).readTree(...)
```